### PR TITLE
feat: Add AWS SSM Parameters resource

### PR DESCRIPTION
**Pull request by spangenberg**
Fixes: https://github.com/cloudquery/cloudquery/issues/1363

### CODE REVIEW

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Looks good but some tests failing

File: plugins/source/aws/resources/services/ssm/parameters_mock_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package ssm
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	"github.com/aws/aws-sdk-go-v2/service/ssm/types"
+	"github.com/cloudquery/cq-provider-aws/client"
+	"github.com/cloudquery/cq-provider-aws/client/mocks"
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/ssm/parameters_mock_test.go:9]:
```suggestion
	"github.com/cloudquery/cloudquery/plugins/source/aws/client"
	"github.com/cloudquery/cloudquery/plugins/source/aws/client/mocks"
```

File: plugins/source/aws/resources/services/ssm/parameters.hcl [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+add_generate     = true
+
+description_modifier "remove_read_only" {
+  words = ["  This member is required"]
+}
+
+resource "aws" "ssm" "parameters" {
+  path = "github.com/aws/aws-sdk-go-v2/service/ssm/types.ParameterMetadata"
+  ignoreError "IgnoreAccessDenied" {
+    path = "github.com/cloudquery/cq-provider-aws/client.IgnoreAccessDeniedServiceDisabled"
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/ssm/parameters.hcl:12]:
```suggestion
    path = "github.com/cloudquery/cloudquery/plugins/source/aws/client.IgnoreAccessDeniedServiceDisabled"
```

File: plugins/source/aws/resources/services/ssm/parameters.hcl [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    type        = "string"
+    resolver "resolveAWSAccount" {
+      path = "github.com/cloudquery/cq-provider-aws/client.ResolveAWSAccount"
+    }
+  }
+  userDefinedColumn "region" {
+    type        = "string"
+    description = "The AWS Region of the resource"
+    resolver "resolveAWSRegion" {
+      path = "github.com/cloudquery/cq-provider-aws/client.ResolveAWSRegion"
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/ssm/parameters.hcl:33]:
```suggestion
      path = "github.com/cloudquery/cloudquery/plugins/source/aws/client.ResolveAWSRegion"
```

File: plugins/source/aws/resources/services/ssm/parameters.hcl [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  multiplex "AwsAccountRegion" {
+    path   = "github.com/cloudquery/cq-provider-aws/client.ServiceAccountRegionMultiplexer"
+    params = ["ssm"]
+  }
+
+  userDefinedColumn "account_id" {
+    description = "The AWS Account ID of the resource"
+    type        = "string"
+    resolver "resolveAWSAccount" {
+      path = "github.com/cloudquery/cq-provider-aws/client.ResolveAWSAccount"
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/ssm/parameters.hcl:26]:
```suggestion
      path = "github.com/cloudquery/cloudquery/plugins/source/aws/client.ResolveAWSAccount"
```

File: plugins/source/aws/resources/services/ssm/parameters.hcl [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+resource "aws" "ssm" "parameters" {
+  path = "github.com/aws/aws-sdk-go-v2/service/ssm/types.ParameterMetadata"
+  ignoreError "IgnoreAccessDenied" {
+    path = "github.com/cloudquery/cq-provider-aws/client.IgnoreAccessDeniedServiceDisabled"
+  }
+  deleteFilter "AccountRegionFilter" {
+    path = "github.com/cloudquery/cq-provider-aws/client.DeleteAccountRegionFilter"
+  }
+  multiplex "AwsAccountRegion" {
+    path   = "github.com/cloudquery/cq-provider-aws/client.ServiceAccountRegionMultiplexer"
```
Comment (erezrokah) [inline plugins/source/aws/resources/services/ssm/parameters.hcl:18]:
```suggestion
    path   = "github.com/cloudquery/cloudquery/plugins/source/aws/client.ServiceAccountRegionMultiplexer"
```

**Pull request merged**